scipy/interpolate/tests/test_polyint.py

```diff
@@ -551,5 +560,57 @@ def test_all_zeros(self):
         assert_equal(pch(xx), 0.)
 
 
+class TestCubicSpline(object):
+    def test_general(self):
+        x = np.array([-1, 0, 0.5, 2, 4, 4.5, 5.5, 9])
+        y = np.array([0, -0.5, 2, 3, 2.5, 1, 1, 0.5])
+
+        x_test = np.array([-0.5, 0.1, 1, 3, 4.1, 5, 7])
+        y_true = np.array([
+            -2.111801336869092, 0.019677035288960, 3.164826064645384,
+            3.384966086626085, 2.210195906837064, 0.464510284120531,
+            4.956649059572205])
```

Comment:
They are from matlab, where "not-a-knot" condition is also used, so it is convenient to test. Fine to keep this test?